lib/puppet/application/ssl.rb

```diff
@@ -91,106 +91,131 @@ def main
       Puppet[:certname] = options[:target]
       Puppet[:confdir]  = File.join(Puppet[:devicedir], Puppet[:certname])
       Puppet[:vardir]   = File.join(Puppet[:devicedir], Puppet[:certname])
-      host = Puppet::SSL::Host.new(Puppet[:certname], true)
       Puppet.settings.use(:main, :agent, :device)
     else
-      host = Puppet::SSL::Host.new(Puppet[:certname])
       Puppet.settings.use(:main, :agent)
     end
 
+    certname = Puppet[:certname]
     action = command_line.args.first
     case action
     when 'submit_request'
-      submit_request(host)
-      cert = download_cert(host)
-      unless cert
-        Puppet.info _("The certificate for '%{name}' has not yet been signed") % { name: host.name }
+      machine = Puppet::SSL::StateMachine.new
+      ssl_context = machine.ensure_ca_certificates
+      if submit_request(ssl_context)
+        cert = download_cert(ssl_context)
+        unless cert
+          Puppet.info(_("The certificate for '%{name}' has not yet been signed") % { name: certname })
+        end
       end
     when 'download_cert'
-      cert = download_cert(host)
+      machine = Puppet::SSL::StateMachine.new
+      ssl_context = machine.ensure_ca_certificates
+      cert = download_cert(ssl_context)
       unless cert
-        raise Puppet::Error, _("The certificate for '%{name}' has not yet been signed") % { name: host.name }
+        raise Puppet::Error, _("The certificate for '%{name}' has not yet been signed") % { name: certname }
       end
     when 'verify'
-      verify(host)
+      verify(certname)
     when 'clean'
-      clean(host)
+      clean(certname)
     else
       raise Puppet::Error, _("Unknown action '%{action}'") % { action: action }
     end
   end
 
-  def submit_request(host)
-    ensure_ca_certificates
+  def submit_request(ssl_context)
+    cert_provider = Puppet::X509::CertProvider.new
+    key = cert_provider.load_private_key(Puppet[:certname])
+    unless key
```

Comment:
Add `Puppet.info _("Creating a new SSL key for %{name}") % { name: Puppet[:certname] }`

Reply:
Fixed in 
ee8a4f0